### PR TITLE
category deleted / product form working / admin middleware not working

### PR DESCRIPTION
**Pull request by Diane-Sayos**

### CODE REVIEW

**Comment by bob-zs** [comment]
🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇

File: db/User.js [hunk 1]
```diff
@@ -141,5 +141,40 @@ User.findByToken = async function findByToken(token) {
...
+User.prototype.updateProduct = async function(productReq, id){
+  let product = await conn.models.product.findByPk(id*1);
+  // product = await conn.models.product.update({
+  //   name: productReq.name,
+  //   description: productReq.description,
+  //   ml: productReq.ml,
+  //   price: productReq.price,
+  //   limit: productReq.limit,
+  //   imgUrl: productReq.imgUrl
+  // });
```
Comment (bob-zs) [inline db/User.js:206]: Let's clean up commented out code. Good job in using the `productReq` for the update. A lot cleaner

File: index.html [hunk 2]
```diff
@@ -81,6 +82,9 @@
     background-color: rgb(235, 188, 188);
     color: black;
   }
+  .side {
+
+  }
```
Comment (bob-zs) [inline index.html]: Please add a style or remove this empty CSS rule

File: routes/products.js [hunk 3]
```diff
@@ -2,44 +2,55 @@ const express = require('express');
...
+  }
+  catch(ex){
+    next(ex)
+  }
+});
+//create a product with authentication
+// //edit/update specific product with authentication
+// app.put('/:id', isLoggedIn, async(req, res, next) => {
+//   try{
+//     res.send(await Product.update(req.body));
```
Comment (bob-zs) [inline routes/products.js:39]: Rm commented out code please.

File: src/App.js [hunk 4]
```diff
@@ -32,6 +33,7 @@ class App extends React.Component {
               <Route exact path='/' component={ UserCart } />
               <Route exact path='/products' component={ ProductsView } />
               <Route exact path='/products/:id' component={ ProductSpecificView } />
+              { auth.isAdmin ? <Route exact path='/products/:id' component={ ProductForm } /> : null }
```
Comment (bob-zs) [inline src/App.js]: 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉

File: src/Header.js [hunk 5]
```diff
@@ -37,16 +37,7 @@ const Header = ({ auth, cart, logout }) => {
...
-              }
-                {/* // <Link to="/" className={ !view ? 'selected' : ''}>Home</Link>
-                // <Link to="/products" className={ view === 'products'? 'selected' : ''}>Products</Link>
-                // <Link to="/cart" className={ view === 'cart'? 'selected' : ''}>Cart ({cart.lineItems.length} items)</Link>
-                // <Link to="/profile" className={ view === 'profile'? 'selected' : ''}>Profile</Link>
-                // <Link to='/wishlist' className={ view === 'wishlist'? 'selected' : ''}>Wish List</Link>
-                // <button onClick={ () => logout(history) }>Logout { auth.username }</button> */}
-            {/* <div>
-                searchbar div
-            </div> */}
```
Comment (bob-zs) [inline src/Header.js]: You are the best! 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇  lol

File: routes/products.js [hunk 6]
```diff
@@ -2,44 +2,55 @@ const express = require('express');
...
-      next(ex);
+  }
+  catch(ex){
+    next(ex)
+  }
+});
+app.put('/:id', isLoggedIn, async(req, res, next) => {
+  try{
+    if(req.user.isAdmin){
+      res.status(201).send(await req.user.updateProduct(req.body, req.params.id));
```
Comment (bob-zs) [inline routes/products.js:28]: Seems like you got the admin checking working 👍

**Comment by bob-zs** [comment]
Seems like it's working
Before update: name of Product "Old Banana Product"
Screen Shot 2022-08-25 at 10 37 38 AM

After update: name of Product "NEW BANANA PRODUCT"
Screen Shot 2022-08-25 at 10 39 35 AM

File: db/User.js (hunk 1 above)
Comment (bob-zs) [inline db/User.js:206]: Please delete commented-out code. Thanks!

File: db/Product.js [hunk 7]
```diff
@@ -16,15 +16,17 @@ const Product = conn.define("product", {
...
   },
-  category: {
-    type: ENUM,
-    values: ["philippines", "china", "korea", "japan", "mexico"],
-  },
+  // category: {
+  //   type: ENUM,
+  //   values: ["philippines", "china", "korea", "japan", "mexico"],
+  // },
   ml: {
```
Comment (bob-zs) [inline db/Product.js:23]: Thinking about keeping this category data? Or for good delete it?

**Review by alejandro-loja — APPROVED** [review]
ok

**Pull request merged**